SPEC.md

```diff
@@ -155,6 +155,37 @@ When persisting an output artifact to a file system, if the build tool has addit
 it should persist that metadata to a subdirectory of the directory to which the output artifact is being written of the form: ```.bom/metadata/${tool}/```.  Filenaming and subdirectory structure below that point is at the discretion of the build tool.
 ## Annex A
 
+## Annex B
+
+### Build tool selection of gitoid type
+
+GITOID_BLOB_TYPE should be set in order of precedence by:
+1. A non-empty environment variable named GITOID_BLOB_TYPE.
+2. A build tool specific flag.
+The value of GITOID_BLOB_TYPE should contain the gitoid type. It may contain multiple gitoid types separated by comma (e.g., sha1,sha256). The build tool should generate the gitoids for all specified types. The build tool should implement a default gitoid type that should be used when no flag is provided.
+The build tool should report an error if the requested gitoid type is not supported.
+
+
+### Gitoid persistence by a Build Tool in ELF Objects/Executables
+When persisting the gitoid to an ELF object or an ELF executable, the build tool should create a section ```.note.gitbom``` of type SHT_NOTE and place the gitoid in the descriptor field of the Note entry. Multiple Note entries should be created, one for each gitoid type when multiple gitoid types are involved. Each Note entry must contain the following fields in the same order as given below:
+1. namesz (4 bytes): This field must be set to a value of 8, the length of the 'owner' field (including padding) in bytes.
+2. descz (4 bytes): This field must contain the length of the gitoid in bytes.
+3. type (4 bytes): This field must contain the value associated with one of the reserved gitoid types or a custom type.
+   The values for the reserved types are in the range of 0x0000 to 0xfffe. Some of the Gitoid types with reserved values are given below:
+   ```{ NT_GITOID_BLOB_SHA1 = 0x0,
+	NT_GITOID_BLOB_SHA224 = 0x1,
+	NT_GITOID_BLOB_SHA256 = 0x2,
+	NT_GITOID_BLOB_SHA384 = 0x3,
```

Comment:
Currently GITOIDS support only sha1 and sha256, what was the thinking beyond specifying a large number of hashes not yet in use?

Reply:
I was inclined to think we should enumerate upfront as many types as possible ! 
I'll limit to these two types (sha1, sha256) for now and we could update the spec when support for other types are added.

Reply:
I have an idea here, not sure if it is good.

Perhaps we can reserve the enum value of 0 for some special meaning? For example, this 0 type means a list of TLVs in the descriptor field, so that a single type=0 note entry will always suffice, no mater how many hashing types in the .note.gitbom section? Or this 0 type means vendor-interpreted descriptor format? Or we can find out a smarter use of this special 0 value in future?

I think it is easier to remember if we have NT_GITOID_BLOB_SHA1=0x1, NT_GITOID_BLOB_SHA256=0x2, which leaves 0 not used for now. :-)

Anyway, just want to see if I can provide inspiration for other people.